gcs/src/main/scala/monix/connect/gcs/Bucket.scala

```diff
@@ -0,0 +1,319 @@
+package monix.connect.gcs
+
+import java.io.InputStream
+import java.nio.file.Path
+
+import cats.data.NonEmptyList
+import com.google.cloud.storage.Bucket.BucketSourceOption
+import com.google.cloud.storage.Storage.{BlobGetOption, BlobListOption, BlobWriteOption, BucketTargetOption}
+import com.google.cloud.storage.{Acl, BlobId, Bucket => GoogleBucket}
+import monix.connect.gcs.configuration.BlobInfo
+import monix.connect.gcs.utiltiies.StorageUploader.StorageConsumer
+import monix.connect.gcs.utiltiies.{FileIO, Paging, StorageDownloader, StorageUploader}
+import monix.eval.Task
+import monix.reactive.Observable
+
+import scala.jdk.CollectionConverters._
+
+/**
+ * This class wraps the [[com.google.cloud.storage.Bucket]] class, providing an idiomatic scala API
+ * handling null values with [[Option]] where applicable, as well as wrapping all side-effectful calls
+ * in [[monix.eval.Task]] or [[monix.reactive.Observable]].
+ *
+ * Example:
+ * {{{
+ *   import monix.reactive.Observable
+ *   import monix.connect.gcs.{Storage, Bucket}
+ *
+ *   val config = BucketConfig(
+ *      name = "mybucket"
+ *   )
+ *
+ *   val storage: Task[Storage] = Storage.create().memoize
+ *   val bucket: Task[Bucket] = storage.flatMap(_.createBucket(config)).memoize
+ *
+ *   (for {
+ *      bucket <- Observable.fromTask(bucket)
+ *      blobs  <- bucket.list()
+ *   } yield println(blob.name)).completeL
+ * }}}
+ */
+final class Bucket private(underlying: GoogleBucket)
```

Comment:
What do you think on renaming this class to `GCPBucket` or similar? I just think that the current import renaming `Bucket => GoogleBucket` could be a bit confusing for the user.

Reply:
I could rename them, I don't capitalize Acronyms as a standard, so can make it `GcsStorage`, `GcsBucket`, `GcsBlob`, `GcsBucketInfo`, and `GcsBlobInfo`